router/repository/memory_router_repository.go

```diff
@@ -38,19 +41,37 @@ type routerRepo struct {
 	takerFeeMap              sync.Map
 	candidateRouteSearchData sync.Map
 
+	baseFeeMx sync.RWMutex
+	baseFee   domain.BaseFee
+
 	logger log.Logger
 }
 
-// New creates a new repository for the router.
 func New(logger log.Logger) RouterRepository {
 	return &routerRepo{
 		takerFeeMap:              sync.Map{},
 		candidateRouteSearchData: sync.Map{},
+		baseFeeMx:                sync.RWMutex{},
+		baseFee:                  domain.BaseFee{},
 
 		logger: logger,
 	}
 }
 
+// GetBaseFee implements RouterRepository.
+func (r *routerRepo) GetBaseFee() domain.BaseFee {
```

Comment:
Router has already a lot of responsibilities, introducing new one has resulted in ripple effect in the code base, notice how many changes in different places was required to make to implement this base fee idea. Additionally, I think router is not good candidate for maintaining `baseFee`, because many other parts in the system will use/benefit of this `baseFee` and each part now will be required to pull `RouterRepository`. 
I would recommend creating instead `fees` ( or similar package ) and move relevant fee logic there, for example:

```
// GetBaseFee ...
func GetBaseFee() sdk.Coin {
	r.baseFeeMx.RLock()
	defer r.baseFeeMx.RUnlock()
	return r.baseFee
}

// SetBaseFee ...
func  SetBaseFee(baseFee sdk.Coin) {
	r.baseFeeMx.Lock()
	defer r.baseFeeMx.Unlock()
	r.baseFee = baseFee
}
```

Using this approach clients would not even need to have any dependency and can call `fees.GetBaseFee()`.

Reply:
I created a separate interface that the router repository extends per your earlier comment so that tx gas calculator depends on a lighter interface.

IMO it is premature to start fragmenting the repository abstractions.

The base fee is used only by the router functionality. As a result, I chose to place it into the router repository